docs/usage/config-migration.md

```diff
@@ -0,0 +1,78 @@
+# Config Migration
+
+Renovate maintainers often need to rename, remove or combine configuration options to improve the user experience and mature the product.
+
+When they do so, "config migration" code is added at the same time so that any "legacy" config from users continues to work.
+Config migration works by migrating legacy config internally before the config is used.
+If done right, it "just works" silently and legacy configs continue working indefinitely.
+The only sign when this is necessary is a debug message in logs noting the old and newly migrated configs.
+
+By default, none of these changes are applied to Renovate config files (e.g. `renovate.json`)
```

Comment:
```suggestion
By default, the "config migration needed" changes are applied at runtime, and then "forgotten".
By default, Renovate will _not_ update the Renovate config file itself (e.g. `renovate.json`).
```

How about something like this? The key point is that by _default_:

- Renovate silently applies the config migrations, and then "forgets about them"
- Renovate does not change/update the config file

Reply:
The "forgotten" word might confuse users. A simple `migrated at runtime` sounds better.